core/src/main/scala/org/http4s/StaticFile.scala

```diff
@@ -129,6 +136,11 @@ object StaticFile {
         .map(isFile =>
           if (isFile) s"${f.lastModified().toHexString}-${f.length().toHexString}" else "")
 
+  def calcETagURL[F[_]: Sync]: URL => F[ETag] = url => {
+    val urlConn = url.openConnection
```

Comment:
Should I be ``F.delay``ing this?  
I only didn't suspend it within an effect because the example on line 93 doesn't do it.  
Isn't ``URL#openConnection`` side-effecting?

Also, would it make more sense to have the signature as ``URLConnection => F[ETag]`` to reuse the same connection?

Reply:
this is blocking, so wrapping in `Sync[F].blocking{}` would make perfect sense

Reply:
Nevermind, I took a look at the docs again and read it more carefully.
Looks like ``URLConnection`` represents a connection that doesn't happen yet.  
Line 109 confirms it for me since ``URLConnection#getInputStream`` is what reads the data.
Now I understand why an ``F.delay`` wasn't needed. It's already a pure value 😂

Reply:
it would be pure unless you where reading headers. That makes it blocking

Reply:
Makes a lot more sense, thanks @hamnis!

Reply:
Maybe this should be private api, since we assume that we will get LastModified And ContentLength?
I assume that this is either a file:// or jar reference?

maybe also this should use `getContentLengthLong` as well.

Reply:
Oh whoops I had it as ``getContentLengthLong`` earlier and forgot to keep it as that when I added those changes.
Good point on making it private though. I also forgot that I'm assuming the url also contains a file from where the ``etagCalculator`` is being called